Signal/src/ViewControllers/ContactViewController.swift

```diff
@@ -388,32 +359,37 @@ class ContactViewController: OWSViewController, CNContactViewControllerDelegate
 //        }
 
         for phoneNumber in contact.phoneNumbers {
-            let formattedPhoneNumber = PhoneNumber.bestEffortLocalizedPhoneNumber(withE164: phoneNumber.phoneNumber)
-
-            rows.append(createNameValueRow(name: phoneNumber.localizedLabel(),
-                                      value: formattedPhoneNumber,
-                                      actionBlock: {
-                                        guard let url = NSURL(string: "tel:\(phoneNumber.phoneNumber)") else {
-                                            owsFail("\(ContactViewController.logTag) could not open phone number.")
-                                            return
-                                        }
-                                        UIApplication.shared.openURL(url as URL)
+            rows.append(ContactFieldView.contactFieldView(forPhoneNumber: phoneNumber,
```

Comment:
Added factory methods to `ContactFieldView` that produce standard, reusable views for contact properties. DRY and ensures consistency of presentation.